src/unraphael/dash/align.py

```diff
@@ -466,3 +469,158 @@ def align_image_to_base(
+    keypoints1 = descriptor_extractor.keypoints
+    descriptors1 = descriptor_extractor.descriptors
+
+    descriptor_extractor.detect_and_extract(target_image_gray)
+    keypoints2 = descriptor_extractor.keypoints
+    descriptors2 = descriptor_extractor.descriptors
+
+    # Check if keypoints are detected
+    if keypoints1.shape[0] == 0 or keypoints2.shape[0] == 0:
+        st.error('No keypoints detected in one of the images.')
+        return
+
+    # Match descriptors
+    matches = match_descriptors(
+        descriptors1, descriptors2, max_ratio=max_ratio, cross_check=True
+    )
+
+    # Check if matches are found
+    if len(matches) == 0:
+        st.error('No matches found between images.')
+        return
+
+    # Plot the matched keypoints with image names as titles
+    fig, ax = plt.subplots(figsize=(10, 8))
+    ax.axis('off')
+
+    if display_in_grayscale:
+        plot_matches(ax, base_image_gray, target_image_gray, keypoints1, keypoints2, matches)
+    else:
+        plot_matches(ax, base_image, target_image, keypoints1, keypoints2, matches)
+
+    # Adding the image names above the corresponding images
+    ax.text(
+        0.25,
+        1.05,
+        f'{base_image_name}',
+        ha='center',
+        va='center',
+        fontsize=12,
+        transform=ax.transAxes,
+    )
+    ax.text(
+        0.75,
+        1.05,
+        f'{target_image_name}',
+        ha='center',
+        va='center',
+        fontsize=12,
+        transform=ax.transAxes,
+    )
+
+    st.pyplot(fig)
+
+
+def feature_alignment_navigation_widget(
+    base_image: ImageType,
+    images: list[ImageType],
+    method: str,
+    display_in_grayscale: bool = True,
+    max_ratio: float = 0.6,
+):
+    """Widget to navigate and display feature-based alignment between the base
+    image and other images.
+
+    Parameters:
+    - base_image: The base image
+    - images: List of aligned images to compare with the base image
+    - method: The feature detection method used (i.e., 'SIFT' or 'ORB')
+    - display_in_grayscale: Whether to display images in grayscale or original color
+    - max_ratio: The maximum ratio for descriptor matching
+    """
+
+    if 'image_index' not in st.session_state:
+        st.session_state.image_index = 0
+
+    def display_current_image():
+        current_image = images[st.session_state.image_index]
+        feature_based_alignment_visual(
+            base_image=base_image.data,
+            target_image=current_image.data,
+            method=method,
+            base_image_name=base_image.name,
+            target_image_name=current_image.name,
+            display_in_grayscale=display_in_grayscale,
+            max_ratio=max_ratio,
+        )
+
+    def next_image():
+        st.session_state.image_index = (st.session_state.image_index + 1) % len(images)
+
+    def previous_image():
+        st.session_state.image_index = (st.session_state.image_index - 1) % len(images)
```

Comment:
Nice to have: We use this pattern now in several places. Can we refactor/generalize it so that we avoid this code duplication everywhere?